libguestfs: fix perl commands

ljli opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change
Commands depending on perl such as `virt-list-partitions` or `virt-list-filesystems` were not working, as some dependencies couldn't be found by perl.

###### Things done
Provide dependencies via `PERL5LIB` wrapper.

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

ljli committed:
libguestfs: fix perl commands

volth commented on pkgs/development/libraries/libguestfs/default.nix:
```diff
@@ -18,13 +18,15 @@ stdenv.mkDerivation rec {
-    numactl xen libapparmor getopt perlPackages.ModuleBuild
-  ] ++ (with perlPackages; [ perl libintl_perl GetoptLong SysVirt ])
+    numactl xen libapparmor getopt
+    perlPackages.perl
```
`perl` is already included via `perlDeps`, so this line can be removed

ljli replied:
Removed `perl` from `perlDeps`.

ljli commented:
I addressed the review comment.
Regarding refactoring, I tried to use `perl.withPackages`. The libguestfs perl commands depend on a perl module path in the derivation itself (second `--prefix PERL5LIB` argument to `makeWrapper`). But `perl.withPackages` produces a wrapped `perl` that simply sets `PERL5LIB`, overwriting the inherited `PERL5LIB` content without adding to it. I wasn't sure how to deal with that.

volth commented:
> But perl.withPackages produces a wrapped perl that simply sets PERL5LIB, overwriting the inherited PERL5LIB content without adding to it. I wasn't sure how to deal with that.

Oh, it seems like a my bug.

volth commented:
@ljli (or me, or anyone who might go the refactor way): besides #78118 it will require #59976-like fix for `hivex`

JohnAZoidberg commented:
Seems `virt-list-partitions` or `virt-list-filesystems` have been replaced: https://libguestfs.org/virt-filesystems.1.html
Is this PR still needed?

JohnAZoidberg committed:
Merge remote-tracking branch 'upstream/master' into fix-libguestfs-perl

JohnAZoidberg commented:
In case we still need it, I merged master into it, to fix the conflicts.

twz123 commented:
I've opened a version update for libguestfs which addresses the issue with the Perl bindings: #185015. Note that as of [libguestfs 1.46](https://github.com/libguestfs/libguestfs/blob/ab2d624f46b6cb69a332f513ea32a1f7c9170ca5/docs/guestfs-release-notes-1.46.pod), many of the CLI tools (including virt-filesystems) have moved to the new [guestfs-tools](https://github.com/rwmjones/guestfs-tools) project. That one is not yet packaged for NixOS, but I'm giving it a stab right now.
